geopandas/tests/test_geoseries.py

```diff
@@ -152,6 +152,17 @@ def test_coord_slice(self):
         self.assertTrue(geom_equals(self.g3[[True, False]], self.g3.cx[0.9:, :0.1]))
         self.assertTrue(geom_equals(self.g3[[False, True]], self.g3.cx[0:0.1, 0.9:1.0]))
 
+    def test_coord_slice_with_zero(self):
+        """Test that CoordinateSlice correctly handles zero slice (#GH477)."""
+
+        gs = GeoSeries([Point(x, x) for x in range(-3, 4)])
+        assert(geom_equals(gs.cx[:0, :0], gs.loc[:3]))
```

Comment:
yeah, we can in principle convert those to plain `assert`'s, but fine to keep it consistent for now.